docs/builtins.md

```diff
@@ -0,0 +1,718 @@
+provided it only prints the newline character. 
+
+A tab is printed for the `\t` control character. A newline is printed
+for the `\n` control character. Quotes in string literals are escaped
+with backslash `\`, for example `print "Here's a tab: \t.\nShe said: \"Thank you!\""`.
+
+In a browser environment `print` typically prints to a DOM Element wired
+to the evy runtime. When using evy as Command Line Interface, `print`
+prints to Stdout.
+
+### `read`
+ 
+`read` reads a line of input from the user and returns it as a
+string. The newline character is not included in the returned string.
+
+#### Example
+```evy
+name := read
+print "Hello, "+name+"!"
+```
+
+Input
+```
+Mary Jackson
+```
+
+Output
+```
+Hello, Mary Jackson!
+```
+
+#### Reference 
+
+    read:string
+
+`read` returns a string that contains the line of input that the user
+entered up until, excluding the newline character. It is a blocking
+functions, which means that it will not return until the user has
+entered a line of input and pressed the Enter key.
+
+In a browser environment `read` typically reads from a DOM Textarea
+Element wired to the evy runtime. When using evy as Command Line
+Interface, `read` reads from Stdin.
+
+### `cls`
+
+`cls` clears the output area of all printed text.
+
+#### Example
+```evy
+print "Hello"
+sleep 1
+cls
+print "Bye"
+```
+
+Output
+```
+Bye
+```
+
+#### Reference 
+
+    cls
+
+In a browser environment `cls` typically clears the DOM Element wired to
+the evy runtime which is used for printing. When using evy as Command
+Line Interface, `cls` clears the terminal, similar to the Unix `clear`
+or Windows `cls` commands.
+
+### `printf`
+
+`printf` stands for print formatted.
+
+`printf` prints its arguments to the output area according to a _format_
+string. The format string is the first argument, and it
+contains _specifiers_. Specifiers are placeholder that start with a
+percent sign `%`. They tell the `printf` function how and where to
+print the remaining arguments inside the format string. The rest of the
+format string is printed to the output area without changes. 
+
+Here are some valid specifiers in Evy:
+
+| Specifier | Description |
+| --------- | ------------- |
+| `%v`      | the argument in its default format  |
+| `%q`      | a double-quoted string |
+| `%%`      | a percent sign `%`  |
+
+#### Example
+
+```evy
+printf "%%v string: %v\n" "abc"
+printf "%%v bool: %v\n" true
+printf "%%v num: %v\n" 1
+print
+printf "%%q string: %q\n" "abc"
+printf "%%v num with decimals: %v\n" 1.234
+print
+printf "%%v array: %v\n" [1 2 3]
+printf "%%v map: %v\n" {a:1 b:2}
+```
+
+Output
+```
+%v string: abc
+%v bool: true
+%v num: 1
+
+%q string: "abc"
+%v num with decimals: 1.234
+
+%v array: [1 2 3]
+%v map: {a:1 b:2}
+```
+
+#### Reference 
+
+    printf format:string a:any...
+
+`printf` prints its arguments to the output area according to
+the _format_ string that is represented by the first argument.
```

Comment:
"that is represented by the first argument is a bit wordy". (I think "represented by" is a longer way of just saying "is". Perhaps

`printf` prints its first argument (the format string) to the output area, replacing
any specifiers in the format string with the following arguments. A _specifier_ starts
with a `%` and specifies how the value of one of the remaining arguments is
formatted. Each specifier (apart from the `%%` specifier, see below) consumes the
next unconsumed argument in order. For example, ...

Reply:
ok, agreed.